File: 2nd Year/1st Semester/MAP/Seminar/Seminar10/Seminar10/Domain/Sarcina.cs
namespace Seminar10.Domain;

public class Sarcina
{
    private string sarcinaId { set; get; }
    private Enum dificultate;
    private int nrOreEstimate { set; get; }

    public Sarcina(Enum dificultate, string sarcinaId, int nrOreEstimate)
    {
        this.dificultate = dificultate;
        this.sarcinaId = sarcinaId;
        this.nrOreEstimate = nrOreEstimate;
    }

    public override string ToString()
    {
        return
            $"{nameof(dificultate)}: {dificultate}, {nameof(sarcinaId)}: {sarcinaId}, {nameof(nrOreEstimate)}: {nrOreEstimate}";
    }
}

File: 2nd Year/1st Semester/MAP/Seminar/Sem13-MAP/src/main/java/ubb/scs/map/sem13map/domain/Sarcina.java
package ubb.scs.map.sem13map.domain;

public class Sarcina extends Entity<String>{
    private Dificultate dificultate;
    private int nrOreEstimate;
    private String denumire;

    public Sarcina(String id, int nrOreEstimate, Dificultate dificultate, String denumire) {
        this.nrOreEstimate = nrOreEstimate;
        this.dificultate = dificultate;
        this.denumire = denumire;
        setId(id);
    }

    public String getDenumire() {
        return denumire;
    }

    public void setDenumire(String denumire) {
        this.denumire = denumire;
    }

    public Dificultate getDificultate() {
        return dificultate;
    }

    public void setDificultate(Dificultate dificultate) {
        this.dificultate = dificultate;
    }

    public int getNrOreEstimate() {
        return nrOreEstimate;
    }

    public void setNrOreEstimate(int nrOreEstimate) {
        this.nrOreEstimate = nrOreEstimate;
    }

    @Override
    public String toString() {
        return denumire + " " + nrOreEstimate + " " + dificultate;
    }
}
